manim/utils/cfgwriter.py

```diff
@@ -0,0 +1,81 @@
+from os import path
+import configparser
+
+from .config_utils import successfully_read_files
+
+from rich.console import Console
+from rich.progress import track
+from rich.color import Color
+
+
+def check_valid_colour(color):
+    """Checks whether the entered color is a valid color according to rich
+    Parameters
+    ----------
+    color : :class:`str`
+        The color to check whether it is valid.
+    Returns
+    -------
+    Boolean
+        Returns whether it is valid color or not.
+    """
+    try:
+        Color.parse(color) is None
+        return True
+    except:
+        return False
+
+
+def main():
+    console = Console()
+    config = configparser.ConfigParser()
+    config.read(
+        successfully_read_files
+    )
+    default = {
+        "logging.keyword": "bold yellow",
+        "logging.level.notset": "dim",
+        "logging.level.debug": "green",
+        "logging.level.info": "blue",
+        "logging.level.warning": "red",
+        "logging.level.error": "red bold",
+        "logging.level.critical": "red bold reverse",
+        "log.level": "",
+        "log.time": "cyan dim",
+        "log.message": "",
+        "log.path": "dim",
+    }
+    console.print(
+        "[yellow bold]Manim Logger Configuration Editor[/yellow bold]", justify="center"
+    )
+    console.print(
+        "[red]The default colour is shown as input Statement.\nIf left empty default value will be assigned.[/red]"
+    )
+    console.print(
+        "[magenta]Please follow the link for available styles.[/magenta][link=https://rich.readthedocs.io/en/latest/style.html]docs[/link]"
+    )
+    for key in default:
+        console.print("Enter the Style for %s" % key + ":", style=key, end="")
+        temp = input()
+        if temp:
+            while check_valid_colour(temp) == False:
+                console.print(
+                    "[red bold]Your Style is not valid. Try again.[/red bold]"
+                )
+                console.print("Enter the Style for %s" % key + ":", style=key, end="")
+                temp = input()
+            else:
+                default[key] = temp
+    config["log.color"] = default
+    for n in track(range(100), description="Converting to Manim.cfg"):
+        with open("manim.cfg", "w") as fp:
+            config.write(fp)
```

Comment:
I'd strongly suggest abstracting these lines into a separate function. Also, are we just writing to the current directory? We should change that, pending #179

Reply:
@leotrs With reference to the cascading config file system that was discussed, aren't we going to have multiple config files that have the capability to override each other? Why should this write _only_ to the places mentioned it #179 ?

I suggest an option could be added where the user can choose the path to save the config to.

Reply:
At the moment, there are three different levels of config files. Library-wide, user-wide, and folder-wide. cfg-writer (nor anything/anyone) should change the first one (library-wide). When ran without arguments, then I think cfg/writer should change the second one (user-wide). If ran with some kind of flag/argument, then cfg-writer could change the third one (folder-wide).

All that just to say: I agree with you @Aathish04 :)

Reply:
Done.